digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectAddressLinkBuilder.scala

```diff
@@ -62,21 +62,27 @@ object ProjectAddressLinkBuilder extends AddressLinkBuilder {
     build(roadLink, 0L, geom, length, roadLinkRoadNumber, roadLinkRoadPartNumber, roadLinkTrackCode, roadName, municipalityCode,
       linkType, UnknownRoadLinkType, Discontinuity.Continuous, missingAddress.startAddrMValue.getOrElse(0), missingAddress.endAddrMValue.getOrElse(0),
       missingAddress.startMValue.getOrElse(0.0), missingAddress.endMValue.getOrElse(0.0),SideCode.Unknown,
-      None, None, Anomaly.None, 0, LinkStatus.Unknown, 0)
+      None, None, Anomaly.None, 0, LinkStatus.Unknown, 0, None)
   }
 
   private def build(roadLink: RoadLinkLike, id: Long, geom: Seq[Point], length: Double, roadNumber: Long, roadPartNumber: Long,
                     trackCode: Int, roadName: String, municipalityCode: Int, linkType: LinkType, roadLinkType: RoadLinkType,
                     discontinuity: Discontinuity,
                     startAddrMValue: Long, endAddrMValue: Long, startMValue: Double, endMValue: Double,
                     sideCode: SideCode, startCalibrationPoint: Option[CalibrationPoint], endCalibrationPoint: Option[CalibrationPoint],
-                    anomaly: Anomaly, lrmPositionId: Long, status: LinkStatus, roadAddressId : Long): ProjectAddressLink = {
+                    anomaly: Anomaly, lrmPositionId: Long, status: LinkStatus, roadAddressId: Long, connectedLinkId: Option[Long]): ProjectAddressLink = {
 
-    ProjectAddressLink(id, roadLink.linkId, geom,
+    val linkId =
+      if (connectedLinkId.nonEmpty && status == LinkStatus.New)
+        0L - roadLink.linkId
```

Comment:
what will happen in this case . Will the linkId be a negative linkId?

Reply:
Yes. There will be two links for the UI within a single link which will be represented by positive and negative numbers.